vpcinfo.go

```diff
@@ -106,7 +106,47 @@ type Zone string
 // String returns z as a string value, satisfies the fmt.Stringer interface.
 func (z Zone) String() string { return string(z) }
 
-func whereAmI() (Platform, error) {
+func zoneFromSubnets(subnets []Subnet) (Zone, error) {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return "", err
+	}
+
+	for _, addr := range addrs {
+		for _, subnet := range subnets {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
```

Comment:
It could help with readability to move this to the outer loop since `addr` doesn't change in the inner loop, I was confused for a bit and thought we were taking a value from `subnet`.

Not a blocker tho :)

Reply:
Fixed!